### DISCUSSION TITLE
How to use WithSpring on useSharedValue({ x: 0, y: 0})

### DISCUSSION THREAD

**Discussion by cloudorbush** [software-mansion/react-native-reanimated#2861] [Q&A] [answered]
I have a useSharedValue with an object like this:

```jsx
const offset = useSharedValue({ x: 1, y: 1 })
```

To manipulate the values of the keys I've tried something like this: 

```jsx
      offset.value = {
        x: withSpring(0, {
          stiffness: 60,
          overshootClamping: true,
        }),
        y: withSpring(0, {
          stiffness: 60,
          overshootClamping: true,
        })
      }
```

But it gives an error.

Is there another approach that can be used?

**Comment by JasonCareter** [comment, marked as answer]
why don't you separete the offset  with offsetx = useSharedValue(1); offsety = useSharedValue(1);
Object Shared Value is mainly used for recording some animation state .

**Reply by cloudorbush** [reply]
Followed your advice and it worked, thanks!

**Reply by Noitidart** [reply]
What was the error. It looks like it should work fine.
